components/integrations/bots/add_bot/add_bot.jsx

```diff
@@ -285,9 +285,26 @@ export default class AddBot extends React.PureComponent {
         } else {
             const usernameError = Utils.isValidBotUsername(bot.username);
             if (usernameError) {
+                if (usernameError.id === 'INVALID_LAST_CHARACTER') {
+                    this.setState({
+                        adding: false,
+                        error: (
+                            <FormattedMessage
+                                id='bots.manage.add.invalid_last_char'
+                                defaultMessage='Bot usernames cannot have period as last character'
+                            />
+                        ),
+                    });
+                    return;
+                }
                 this.setState({
                     adding: false,
-                    error: usernameError,
+                    error: (
+                        <FormattedMessage
+                            id='bots.manage.add.invalid_username'
+                            defaultMessage='Usernames have to begin with a lowercase letter and be 3-22 characters long. You can use lowercase letters, numbers, periods, dashes, and underscores.'
+                        />
+                    ),
```

Comment:
Depending on the error we could create a appropriate `FormattedMessage` first and then call the `this.setState` method.

Reply:
done